src/transport_ofi.c

```diff
@@ -591,9 +591,16 @@ static inline int atomic_limitations_check(void)
 
 static inline int exchange_and_av_insert(struct fabric_info *info)
 {
-    int                 i, rank = 0, ret = 0;
-    char   epname[128], *alladdrs = NULL;
-    size_t epnamelen = sizeof(epname);
+    int    i, rank = 0, ret = 0;
+    size_t epnamelen = 128, full_ep_len = epnamelen + HOST_NAME_MAX, adrlen = 0;
```

Comment:
Is ```HOST_NAME_MAX``` portable?  It looks like you may need to add 1 for the NULL byte.

MacOS doesn't define ```HOST_NAME_MAX```.  The manpage says you are supposed to use ```_POSIX_HOST_NAME_MAX``` +1 for NULL instead.